File: packages/ts-pom-bdd/pages/components/HeaderComponent.ts
import { Page, Locator } from '@playwright/test';

export class HeaderComponent {
  private readonly loginButton: Locator;
  private readonly logoutButton: Locator;
  private readonly enButton: Locator;
  private readonly jpButton: Locator;
  private readonly userNameDisplay: Locator;

  constructor(private page: Page) {
    this.loginButton = page.getByRole('button', { name: 'ログイン' });
    this.logoutButton = page.getByRole('button', { name: 'ログアウト' });
    this.enButton = page.getByRole('button', { name: 'EN' });
    this.jpButton = page.getByRole('button', { name: 'JP' });
    this.userNameDisplay = page.locator('[class*="user-name"]');
  }

  async clickLogin() {
    await this.loginButton.click();
  }

  async clickLogout() {
    await this.logoutButton.click();
  }

  async switchToEnglish() {
    await this.enButton.click();
  }

  async switchToJapanese() {
    await this.jpButton.click();
  }

  async isLoggedIn() {
    return await this.logoutButton.isVisible();
  }

  async getLoggedInUserName() {
    if (await this.isLoggedIn()) {
      return await this.userNameDisplay.textContent();
    }
    return null;
  }
}

File: packages/java-pom-bdd/src/test/java/com/example/pages/components/HeaderComponent.java
package com.example.pages.components;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

public class HeaderComponent {
    private final Page page;
    private final Locator loginButton;
    private final Locator logoutButton;
    private final Locator enButton;
    private final Locator jpButton;

    public HeaderComponent(Page page) {
        this.page = page;
        this.loginButton = page.getByRole(com.microsoft.playwright.options.AriaRole.BUTTON,
                new Page.GetByRoleOptions().setName("ログイン"));
        this.logoutButton = page.getByRole(com.microsoft.playwright.options.AriaRole.BUTTON,
                new Page.GetByRoleOptions().setName("ログアウト"));
        this.enButton = page.getByRole(com.microsoft.playwright.options.AriaRole.BUTTON,
                new Page.GetByRoleOptions().setName("EN"));
        this.jpButton = page.getByRole(com.microsoft.playwright.options.AriaRole.BUTTON,
                new Page.GetByRoleOptions().setName("JP"));
    }

    public void clickLogin() {
        loginButton.click();
    }

    public void clickLogout() {
        logoutButton.click();
    }

    public boolean isLoginButtonVisible() {
        return loginButton.isVisible();
    }

    public boolean isLogoutButtonVisible() {
        return logoutButton.isVisible();
    }

    public void switchToEnglish() {
        enButton.click();
    }

    public void switchToJapanese() {
        jpButton.click();
    }
}
